### PR TITLE
[cc2538] add support for CC2538-CC2592-EM

### PR DESCRIPTION
**Pull request by sjlongland**
This adds support for the CC2538-CC2592 EM board from TI which is basically identical to the standard CC2538 board, but has an additional CC2592 front-end to boost the transmit power and receive sensitivity.

### CODE REVIEW

File: examples/platforms/cc2538/platform-cc2538.h [hunk 1]
```diff
@@ -90,4 +90,18 @@ void cc2538RandomInit(void);
...
+ * Change the state of the CC2592 HGM pin.
+ *
+ * @param	aState	Whether or not to enable HGM
+ */
+void cc2538RadioSetHgm(bool aState);
+
+/**
+ * Retrieve the state of the CC2592 HGM pin.
+ */
+bool cc2538RadioGetHgm();
```
Comment (jwhui) [inline examples/platforms/cc2538/platform-cc2538.h]: nit: "All void functions or methods shall explicitly declare and specify the void type keyword."

File: examples/platforms/cc2538/radio.c [hunk 2]
```diff
@@ -277,9 +349,60 @@ void cc2538RadioInit(void)
...
+        HWREG_ARR(OPENTHREAD_CONFIG_CC2592_HGM_PORT, GPIO_PIN(OPENTHREAD_CONFIG_CC2592_HGM_PIN)) =
+            GPIO_PIN(OPENTHREAD_CONFIG_CC2592_HGM_PIN);
+    }
+    else
+    {
+        HWREG_ARR(OPENTHREAD_CONFIG_CC2592_HGM_PORT, GPIO_PIN(OPENTHREAD_CONFIG_CC2592_HGM_PIN)) = 0;
+    }
+}
+
+bool cc2538RadioGetHgm()
```
Comment (jwhui) [inline examples/platforms/cc2538/radio.c]: nit: All void functions or methods shall explicitly declare and specify the void type keyword.

Comment (sjlongland) [reply examples/platforms/cc2538/radio.c]: Ahh, too used to languages where `foo()` means a function that takes no arguments.  (or ones that don't care like Perl and JavaScript where they're accessed through `@_` or `arguments` respectively.)

Hopefully fixed in 2662163.

File: examples/platforms/cc2538/openthread-core-cc2538-config.h [hunk 3]
```diff
@@ -93,4 +93,102 @@
  */
 #define SETTINGS_CONFIG_PAGE_SIZE                              2048
 
+/**
+ * @def Enable support for the CC2592 range-extender front-end.
```
Comment (jwhui) [inline examples/platforms/cc2538/openthread-core-cc2538-config.h]: The `@def` command should be followed by the macro that is being documented.

For example:
```c
/**
 * @def OPENTHREAD_CONFIG_CC2538_WITH_CC2592
 *
 * Enable support for the CC2592 range-extender front-end.
 *
 * This is a feature of the CC2538-CC2592 EM and other peripherals which
 * extends the range of the bare CC2538 to over a kilometre line-of-sight.
 * The CC2592 needs to be wired up to the RF port on the CC2538 in accordance
 * with application note 130 ("Using CC2592 Front End With CC2538", TI doc
 * SWRA447).
 *
 */
```

I see that some of the existing doxygen above makes the same mistake.

Comment (sjlongland) [reply examples/platforms/cc2538/openthread-core-cc2538-config.h]: Yeah, monkey-see-monkey-do in my case… I shall fix it.

Comment (jwhui) [reply examples/platforms/cc2538/openthread-core-cc2538-config.h]: Yes, there are a number of similar mistakes across the `examples/`. However, `include/` and `src/` seem generally consistent (see [src/core/openthread-core-default-config.h](https://github.com/openthread/openthread/blob/master/src/core/openthread-core-default-config.h)).

As an example:
```c
/**
 * @def OPENTHREAD_CONFIG_STACK_VENDOR_OUI
 *
 * The Organizationally Unique Identifier for the Thread stack.
 *
 */
#ifndef OPENTHREAD_CONFIG_STACK_VENDOR_OUI
#define OPENTHREAD_CONFIG_STACK_VENDOR_OUI 0x18b430
#endif
```

Comment (sjlongland) [reply examples/platforms/cc2538/openthread-core-cc2538-config.h]: Ahh, gotcha … should be fixed now in 50d69ac8.

**Comment by sjlongland** [comment]
Small errata issue with this set of patches… it appears the RSSI readings are about 15dB higher than reality as I've gotten RSSI offset and RSSI sensitivity muddled.
I'll submit a new PR to address this.

**Pull request merged**